File: Buildings.java
/**
 * Abstract class which every building extends from
 */
public abstract class Buildings extends Quacks {
    private final int BASE_COST;
    private final double MULTIPLICITY; // value that cost of building increases by (ex. 15% or 0.15)
    private double qps; // quacks per second
    private double qpc; // quacks per click
    private int counter; //counts # of buildings
    
    public Buildings(double multiplier, int base) {
        super();
        BASE_COST = base;
        MULTIPLICITY = multiplier;
        counter++;
    }

    /**
     * Takes building count and raises 1.15 to the power of building count. Returns the result
     * 
     */
    public double multiplier(int building_count) {
        double multiplier = Math.pow(1.15, building_count);
        return multiplier;
    }
}

File: qcActual/Buildings.js
/**
 * Base Building class that all building types can extend from.
 */
class Building extends Quacks {
  static MULTIPLIER = 1.15; // cost increases by 15% per additional building
  constructor(cost, qps, qpc) {
    super();
    this.cost = cost;
    this.baseQPS = qps;
    this.baseQPC = qpc;
    this.qps = 0;  // quacks per second
    this.qpc = 0;  // quacks per click
    this.amount = 0;
  }

  getAmt() {
    return this.amount;
  }
  getQPS() {
      return this.qps;
  }
  getQPC() {
    return this.qpc;
  }
  getCost() {
    return this.cost;
  }
  getBaseQPS() {
    return this.baseQPS;
  }
  getBaseQPC() {
    return this.baseQPC;
  }
  buyBuilding() {
      this.amount++;
      quacks.removeQuacks(this.cost);
      this.cost *= this.MULTIPLIER;

      if (this.qps = 0) {
        this.qps = this.baseQPS;
      }
      else {
        this.qps = this.baseQPS * this.amount;
      }
      
      if (this.qpc = 0) {
        this.qpc = this.baseQPC;
      }
      else {
        this.qpc = this.baseQPC * this.amount;
      }
  }
}
